compiler/ccgstmts.nim

```diff
@@ -834,8 +834,12 @@ proc genTryCpp(p: BProc, t: PNode, d: var TLoc) =
       endBlock(p)
     else:
       for j in 0..t[i].len-2:
-        assert(t[i][j].kind == nkType)
-        startBlock(p, "catch ($1*) {$n", getTypeDesc(p.module, t[i][j].typ))
+        if t[i][j].isInfixAs():
+          let v = rope($t[i][j][2])
+          fillLoc(t[i][j][2].sym.loc, locTemp, t[i][j][2], v, OnUnknown)
```

Comment:
Some name for ``t[i][j][2]`` would be nice. (Yes, I think I know what it is.)

Reply:
done